solr/core/src/java/org/apache/solr/core/CoreDescriptor.java

```diff
@@ -64,7 +63,7 @@ public class CoreDescriptor {
   public static final String SOLR_CORE_PROP_PREFIX = "solr.core.";
 
   public static final String DEFAULT_EXTERNAL_PROPERTIES_FILE =
-      "conf" + File.separator + "solrcore.properties";
+      Path.of("conf/solrcore.properties").toString();
```

Comment:
is there any chance of different file seperators?   I wasn't sure if that was one reason for using the `File.separator`, like on windows?  Or is that one of the benefits of Path?

Reply:
I was worried about this so started reading `Path.of()` [documentation](https://github.com/openjdk/jdk/blob/62a4544bb76aa339a8129f81d2527405a1b1e7e3/src/java.base/share/classes/java/nio/file/Path.java#L147) and [oracle documentation](https://docs.oracle.com/javase/tutorial/essential/io/pathOps.html#:~:text=Returns%20the%20string%20representation%20of%20the%20Path). `Filesystems.getDefault().getPath(String)` which should return the OS path and developers can use the standard `/` regardless. It's also a benefit of Path for Solr to no longer have to do all these File.separator chars or OS checking...

I haven't actually tested this on something like a windows operating system to confirm though...